sound/soc/sof/intel/hsw.c

```diff
@@ -167,6 +169,13 @@ static int hsw_run(struct snd_sof_dev *sdev)
 	snd_sof_dsp_update_bits_unlocked(sdev, HSW_DSP_BAR, SHIM_CSR,
 					 SHIM_CSR_STALL, 0x0);
 
+	/* now wait for the DSP to boot */
+	ret = sof_boot_wait_timeout(sdev);
+	if (ret < 0) {
+		dev_err(sdev->dev, "error: fw boot timeout\n");
+		return ret;
+	}
+
 	return 0; //TODO: Fix return value
```

Comment:
I really don't like duplicated code in multiple places. Could we use a util function and just do a return sof_boot_util (find a better name)